pkgs/desktops/gnome-3/extensions/buildGnomeExtension.nix

```diff
@@ -0,0 +1,42 @@
+{ pkgs, lib, stdenv, fetchurl, gnome3, unzip }:
+
+let
+  buildGnomeExtension = {
+    # Every gnome extension has a UUID. It's the name of the extension folder once unpacked
+    # and can always be found in the metadata.json of every extension.
+    uuid,
+    name,
+    pname,
+    description,
+    # extensions.gnome.org extension URL
+    link,
+    version,
+    sha256,
+  }:
+
+  stdenv.mkDerivation {
+    inherit pname;
+    version = builtins.toString version;
+    src = fetchurl {
+      url = "https://extensions.gnome.org/extension-data/${
+          builtins.replaceStrings [ "@" ] [ "" ] uuid
+        }.v${builtins.toString version}.shell-extension.zip";
+      inherit sha256;
+    };
+    nativeBuildInputs = [ unzip ];
```

Comment:
I think this input can be avoided with fetchzip.

Reply:
IIRC, the motivation for this was that I need the hash of the zipped tar ball, but I'm not sure anymore. I'll try it out.

Reply:
Weird. The hash error went away with `recursiveHash = false;` and `stripRoot = false;`. But even then I get some weird error: "output path '/nix/store/fd96ivjnxq987svbl0grl57ys3qbibyq-source' should be a non-executable regular file". I'll thus leave it how it is at the moment.